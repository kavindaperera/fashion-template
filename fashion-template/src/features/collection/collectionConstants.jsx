export const FETCH_PRODUCTS = "FETCH_PRODUCTS";
export const FETCH_SUBITEMS = "FETCH_SUBITEMS";
export const GET_CURRENCY = "GET_CURRENCY";
export const GET_VARAINTS = "GET_VARAINTS";