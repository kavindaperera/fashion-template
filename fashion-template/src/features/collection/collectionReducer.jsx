import { createReducer } from "../../app/common/util/reducerUtil";
import { FETCH_PRODUCTS } from "./collectionConstants";
import { firestoreReducer } from 'redux-firestore'

/*const initialState = firestoreReducer.ordered.products;*/
const initialState = [];

export const fetchProducts = (state, payload) => {
  return payload.products
}

export default createReducer(initialState, {
    [FETCH_PRODUCTS]: fetchProducts
})
