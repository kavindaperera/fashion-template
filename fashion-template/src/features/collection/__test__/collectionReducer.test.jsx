import * as types from '../collectionConstants';
import reducer from '../collectionReducer'
import { initialState  } from '../collectionReducer'

describe('collection reducer', () => {
    it('should return the initial state', () => {
      expect(reducer(undefined, {})).toEqual(
        {
          symbol: '$',
        }
      )
    })
    it('should handle GET_CURRENCY', () => {
        expect(
          reducer(initialState , {
            type: types.GET_CURRENCY,
            payload: {
                symbol: '€',
              }
          })
        ).toEqual({
            symbol: '€',
          })

      })
      it('should handle GET_CURRENCY snapshot', () => {
        expect(
          reducer(initialState , {
            type: types.GET_CURRENCY,
            payload: {
                symbol: '€',
              }
          })
        ).toMatchSnapshot()
      })
})