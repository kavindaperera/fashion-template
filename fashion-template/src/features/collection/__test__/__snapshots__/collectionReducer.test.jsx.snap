// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collection reducer should handle GET_CURRENCY snapshot 1`] = `
Object {
  "symbol": "â‚¬",
}
`;
